### PR TITLE
LPS-133194 role table reference definition test

### PR DESCRIPTION
**Pull request by ctang3**
https://issues.liferay.com/browse/LPS-133194

### CODE REVIEW

File: modules/apps/portal-security/portal-security-permission-impl/src/main/java/com/liferay/portal/security/permission/internal/change/tracking/spi/reference/RoleTableReferenceDefinition.java [hunk 1]
```diff
@@ -39,7 +39,10 @@ public void defineChildTableReferences(
 			childTableReferenceInfoBuilder) {
 
 		childTableReferenceInfoBuilder.systemEventReference(
-			RoleTable.INSTANCE.roleId, Role.class);
+			RoleTable.INSTANCE.roleId, Role.class
+		).resourcePermissionReference(
```
Comment (Preston-Crary) [inline modules/apps/portal-security/portal-security-permission-impl/src/main/java/com/liferay/portal/security/permission/internal/change/tracking/spi/reference/RoleTableReferenceDefinition.java:43]: It's a small thing but please flip the order of these. We always have resourcePermissionReference() before systemEventReference().

Comment (ctang3) [reply modules/apps/portal-security/portal-security-permission-impl/src/main/java/com/liferay/portal/security/permission/internal/change/tracking/spi/reference/RoleTableReferenceDefinition.java]: Flipped the order and updated the branch. I'll keep in mind to check for order in the future. Is it generally alphabetical order?

**Comment by Preston-Crary** [comment]
ci:forward

**Comment by Preston-Crary** [comment]
ci:forward

**Comment by ctang3** [comment]
@Preston-Crary picked the TeamTableReferenceDefinitionTest onto this branch from https://github.com/Preston-Crary/liferay-portal/pull/1368 since it needs the RoleTableReferenceDefinition change from https://github.com/Preston-Crary/liferay-portal/pull/1367/commits/0ec4d2fe26e9e1287da3934bba3a40655d667a5a

**Comment by Preston-Crary** [comment]
ci:test:relevant

**Comment by Preston-Crary** [comment]
ci:test:sf

**Comment by Preston-Crary** [comment]
ci:forward

**Pull request closed without merging**